cfme/automate/import_export.py

```diff
@@ -78,38 +120,150 @@ def import_domain_from(self, branch=None, tag=None):
+        query = (
+            self.appliance.db.client.session.query(
+                namespaces.id,
+                namespaces.name,
+                namespaces.description,
+                git_repositories.url,
+                namespaces.ref_type,
+                namespaces.ref,
+            )
+            .filter(namespaces.parent_id == none, namespaces.source == "remote")
             .join(git_repositories, namespaces.git_repository_id == git_repositories.id)
-        for id, name, description, url, git_type, git_type_value in query:
-            if url != self.url:
+        )
+
+        for db_id, name, description, repo_url, git_type, git_type_value in query:
+            if self.url != repo_url:
                 continue
             if not (
-                    git_type == 'branch' and branch == git_type_value or
-                    git_type == 'tag' and tag == git_type_value):
+                git_type == "branch"
+                and branch == git_type_value
+                or git_type == "tag"
+                and tag == git_type_value
+            ):
                 continue
+
             # We have the domain
-            dc = self.appliance.collections.domains
-            return dc.instantiate(
-                db_id=id, name=name, description=description, git_checkout_type=git_type,
-                git_checkout_value=git_type_value)
+            return self.appliance.collections.domains.instantiate(
+                db_id=db_id,
+                name=name,
+                description=description,
+                git_checkout_type=git_type,
+                git_checkout_value=git_type_value,
+            )
         else:
-            raise ValueError('The domain imported was not found in the database!')
+            raise ValueError("The domain imported was not found in the database!")
+
+
+@attr.s
+class AutomateFileImport(BaseEntity):
+    file_path = attr.ib()
+
+    def import_domain_from(self, domain_from, domain_into=None, toggle_all=True):
+        """Import the domain from file using the Import/Export UI.
+
+        Args:
+            domain_from: Select domain you wish to import from
+            domain_into: Select existing domain to import into:
+            toggle_all: Select all namespaces
+
+        Returns:
+            Instance of :py:class:`cfme.automate.explorer.domain.Domain`
+        """
+        view = navigate_to(self.parent, "All")
+        view.import_file.upload_file.fill(self.file_path)
+        view.import_file.upload.click()
+
+        file_view = self.create_view(FileImportSelectorView, wait="10s")
+        file_view.fill(
+            {"import_into": domain_into, "import_from": domain_from, "toggle_all": toggle_all}
+        )
+        file_view.commit.click()
+        return self.appliance.collections.domains.instantiate(name=domain_from)
+
+
+@attr.s
+class AutomateImportExportsCollection(BaseCollection):
+    def instantiate(
+        self,
```

Comment:
such formatting makes me sad. it kills readability and looks like just waste of space

Reply:
I tried :D. I forgot `black` checks ha ha...